tensorflow/python/ops/math_grad.py

```diff
@@ -411,6 +417,16 @@ def _SigmoidGrad(op, grad):
     return gen_math_ops._sigmoid_grad(y, grad)
 
 
+@ops.RegisterGradient("SigmoidGrad")
+def _SigmoidGradGrad(op, grad):
+  y = op.outputs[0]
+  with ops.control_dependencies([grad.op]):
+    if y.dtype.is_complex:
+      y = math_ops.conj(y)
+    return (- math.ops.exp(x) * (math.ops.exp(x) - 1) *
+            math_ops.pow(math.ops.exp(x) + 1, -3))
```

Comment:
This explodes for large `x` and has the downside of being a constant function of grad (unusual for a gradient, so maybe a test is in order).  Instead, do

    s = math_ops.sigmoid(x)
    return (grad * s) * (1 - s) * (1 - 2 * x)